src/BaseDAO/CLI.hs

```diff
@@ -121,21 +124,49 @@ tzip16MetadataParser = do
       mkCLOptionParser Nothing (#name .! "metadata-key")
         (#help .! "Key in metadata host contract where baseDAO's metadata is located")
 
+-- | Parse metadata for token with given name and given default values.
+tokenMetadataParser
+  :: String
+  -> Text
+  -> Text
+  -> Text
+  -> Opt.Parser FA2.TokenMetadata
+tokenMetadataParser prefix defSymbol defName defDecimals = do
+  symbol <-
+    mkCLOptionParser (Just defSymbol) (#name .! (prefix <> "-token-symbol"))
+    (#help .! "Symbol of the token (according to TZIP-12)")
+  name <-
+    mkCLOptionParser (Just defName) (#name .! (prefix <> "-token-name"))
+    (#help .! "Name of the token (according to TZIP-12)")
+  decimals <-
+    mkCLOptionParser (Just defDecimals) (#name .! (prefix <> "-token-decimals"))
+    (#help .! "Decimals field of the token (according to TZIP-12)")
+  return $ FA2.mkTokenMetadata name symbol decimals
```

Comment:
Meh, I am quite surprised that `mkTokenMetadata` accepts any `Text` as `decimals`, but it's our of scope of this issue. I think we should reject any `decimals` value that is not a number, but AFAIU currently we don't.

Reply:
So I propose to parse it as a number and then convert to `Text`.

Reply:
Ah okay, I decided that there is a special reason why any `Text` is allowed. Let's really parse as numbers.